lib/ramble/ramble/application.py

```diff
@@ -2216,88 +2221,111 @@ def _analysis_dicts(self, criteria_list):
             if log_path in files:
                 files[log_path]["success_criteria"].append(criteria.name)
 
-        # Remap fom / context / file data
         # Could push this into the language features in the future
-        fom_definitions = self.figures_of_merit.copy()
-        for fom, fom_def in fom_definitions.items():
-            fom_def["origin"] = self.name
-            fom_def["origin_type"] = "application"
-
-        fom_contexts = self.figure_of_merit_contexts.copy()
+        fom_sources = [self]
         for mod in self._modifier_instances:
-            fom_contexts.update(mod.figure_of_merit_contexts)
+            fom_sources.append(mod)
+        if self.workflow_manager is not None:
+            fom_sources.append(self.workflow_manager)
+
+        all_contexts = {}
+        for source in fom_sources:
+            for when_fs, source_context_defs in source.figure_of_merit_contexts.items():
+                if self.expander.satisfies(list(when_fs), variant_set=self.object_variants):
```

Comment:
It might be good to push the conversion of `when_fs` -> `when_list` into the `satisfies` method, just to make it a bit more clean.

I was also thinking about pushing `str` -> `[str]` into that so you could call it on a single string and not have it break.